common_tags.py

```diff
@@ -0,0 +1,11 @@
+from __future__ import annotations
+
+
+def find_common_tags(articles: list[dict[str, list[str]]]) -> set[str]:
+    if not articles:
+        return set()
+
+    common_tags = articles[0]["tags"]
+    for article in articles[1:]:
```

Comment:
```suggestion
    common_tags = set(articles[0]["tags"])
        common_tags.intersection_update(article["tags"])
    return common_tags
```

Reply:
`from __future__ import annotations


def find_common_tags(articles: list[dict[str, list[str]]]) -> set[str]:
    if not articles:
        return set()

    common_tags = articles[0]["tags"]
    for article in articles[1:]:
        common_tags = [tag for tag in common_tags if tag in article["tags"]]
    return set(common_tags)
    #this is great`,

    // Improved code you want to suggest
    newContent: `from __future__ import annotations


def find_common_tags(articles: list[dict[str, list[str]]]) -> set[str]:
    if not articles:
        return set()

    common_tags = set(articles[0]["tags"])
    for article in articles[1:]:
        common_tags.intersection_update(article["tags"])
    return common_tags
    #this is great`,